spec/migrations/umm_c_15_4_to_15_5_spec.rb

```diff
@@ -0,0 +1,205 @@
+require File.join(Rails.root, 'db', 'migrate', '20201221134953_umm_c_15_4_to_15_5')
+
+# Testing cases where: 
+# 0. No GetData or Format keys      (both)
+# 1. same but different case
+  # 1.1 same but upcase             (only for up migration) ascii
+  # 1.2 same but downcase           (only for down migration) ascii
+  # 1.3 same but capitalize         (only for up migration) binary
+  # 1.4 same but weirdcase          (only for up migration) geotiff
+# 2. different                      (only for down migration)
+# 3. equivalent HDF4                (both)
+
+describe 'Migration tests for UMM-C v1.15.4 => 1.15.5' do
+  let(:getdata_hash)                       { {'Size' => 42, 'Unit' => 'KB'} }
+  let(:getdata_hash_binary_format_up)      { getdata_hash.merge('Format' => 'binary')  }
+  let(:getdata_hash_geotiff_format_up)     { getdata_hash.merge('Format' => 'geotiff') }
+  let(:getdata_hash_hdf4_format)           { getdata_hash.merge('Format' => 'HDF4')    }
+  let(:getdata_hash_binary_format_down)    { getdata_hash.merge('Format' => 'Binary')  }
+  let(:getdata_hash_geotiff_format_down)   { getdata_hash.merge('Format' => 'GeoTIFF') }
+  let(:getdata_hash_different_format_down) { getdata_hash.merge('Format' => 'GeoPackage') }
+
+
+  context 'when doing the migration for MMT' do
+    before do
+      @draft = create(:full_collection_draft)
+      @template = create(:full_collection_template)
+
+      draft = @draft.draft
+      base_related_url = draft.dig('RelatedUrls',1).clone
+      draft['RelatedUrls'][2] = base_related_url.merge('GetData' => getdata_hash_binary_format_up)
+      draft['RelatedUrls'][3] = base_related_url.merge('GetData' => getdata_hash_geotiff_format_up)
+      draft['RelatedUrls'][4] = base_related_url.merge('GetData' => getdata_hash_hdf4_format)
+      draft['RelatedUrls'][5] = base_related_url.merge('GetData' => getdata_hash_binary_format_down)
+      draft['RelatedUrls'][6] = base_related_url.merge('GetData' => getdata_hash_geotiff_format_down)
+      draft['RelatedUrls'][7] = base_related_url.merge('GetData' => getdata_hash_different_format_down)
+      @draft.save
+
+      template = @template.draft
+      template['RelatedUrls'][2] = base_related_url.merge('GetData' => getdata_hash_binary_format_up)
+      template['RelatedUrls'][3] = base_related_url.merge('GetData' => getdata_hash_geotiff_format_up)
+      template['RelatedUrls'][4] = base_related_url.merge('GetData' => getdata_hash_hdf4_format)
+      template['RelatedUrls'][5] = base_related_url.merge('GetData' => getdata_hash_binary_format_down)
+      template['RelatedUrls'][6] = base_related_url.merge('GetData' => getdata_hash_geotiff_format_down)
+      template['RelatedUrls'][7] = base_related_url.merge('GetData' => getdata_hash_different_format_down)
+      @template.save
+    end
+
+    after do
```

Comment:
I am surprised it is necessary to have this after block to remove the records. I would think rspec would handle that for us.

Reply:
I just followed what I saw a few other specs do. Haven't really done any research on it, but I can take a look and maybe make a low-priority ticket to clear out after blocks that do this.